File: Warehousing.Data/Entities/StoreTransfer.cs
namespace Warehousing.Data.Entities
{
    public class StoreTransfer : BaseClass
    {
        public int Id { get; set; }
        public DateTime TransferDate { get; set; }
        public string Notes { get; set; } = string.Empty;

        // From/To stores
        public int FromStoreId { get; set; }
        public Store FromStore { get; set; } = null!;
        
        public int ToStoreId { get; set; }
        public Store ToStore { get; set; } = null!;

        // Status tracking
        public int StatusId { get; set; }  // Draft, In-Transit, Completed, Cancelled
        public Status Status { get; set; } = null!;

        // Navigation properties
        public ICollection<StoreTransferItem> Items { get; set; } = new List<StoreTransferItem>();
        public ICollection<InventoryTransaction> Transactions { get; set; } = new List<InventoryTransaction>();
    }
}

File: Warehousing.UI/src/app/admin/models/storeTransfer.ts
import { Store } from "./store";
import { Status } from "./status";

export interface StoreTransfer {
    id: number;
    transferDate: string;
    notes: string;
    fromStoreId: number;
    fromStore: Store;
    toStoreId: number;
    toStore: Store;
    statusId: number;
    status: Status;
    items: StoreTransferItem[];
    createdAt: string | null;
    createdBy: string | null;
    updatedAt: string | null;
    updatedBy: string | null;
}

export interface StoreTransferDto {
    id: number | null;
    transferDate: string;
    notes: string;
    fromStoreId: number;
    toStoreId: number;
    statusId: number;
    items: StoreTransferItemDto[];
}

export interface StoreTransferItem {
    id: number;
    transferId: number;
    transfer: StoreTransfer;
    productId: number;
    product: any; // Product interface
    quantity: number;
    unitCost: number;
    notes: string;
    createdAt: string | null;
    createdBy: string | null;
    updatedAt: string | null;
    updatedBy: string | null;
}

export interface StoreTransferItemDto {
    id: number | null;
    transferId: number | null;
    productId: number;
    quantity: number;
    unitCost: number;
    notes: string;
}
